File: leetcode/20.valid-parentheses/cpp/isvalid_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "isvalid.cpp"

struct testCase
{
    std::string s;
    bool want;
};

TEST_CASE("IsValid")
{
    std::vector<testCase>
        tcs{
            {"()", true},
            {"()[]{}", true},
            {"(]", false},
            {"([])", true},
            {"([)]", false},
        };

    Solution s;

    for (auto tc : tcs)
    {
        auto got = s.isValid(tc.s);
        CHECK(got == tc.want);
    }
}

File: leetcode/20.valid-parentheses/ts/isvalid.test.ts
import { test, expect } from "vitest"
import isValid from "./isvalid.js"

interface testcase {
    s: string
    want: boolean
}

test("isvalid", () => {
    const tcs: testcase[] = [
        { s: "()", want: true },
        { s: "()[]{}", want: true },
        { s: "(]", want: false },
        { s: "([])", want: true },
        { s: "([)]", want: false },
    ]

    tcs.forEach(tc => isValid(tc.s) === tc.want)
})
